src/file/file.repository.ts

```diff
@@ -29,52 +32,105 @@ export default class FileRepository {
   }
 
   /**
-   * Updates the file metatata by its id.
+  * Add a given shortcut file to the DB.
+  * @param file - is the shortcut file to be added to the DB
+  */
+  static async createShortcut(file: any): Promise<IFile> {
+    const populatedShortcut = await shortcutModel.create(file);
+    const shortcutAsFile: IFile = this.populatedShortcutToFile(populatedShortcut);
+
+    return shortcutAsFile;
+  }
+
+  /**
+   * Update a file metatata by its id.
    * @param id - the file id.
    * @param partialFile - the partial file containing the attributes to be changed.
    */
   static async updateById(id: string, partialFile: Partial<IFile>): Promise<boolean> {
-    const file = await fileModel.findById(id);
+    const baseFile = await baseFileModel.findById(id);
+    const file = await this.baseFileToIFile(baseFile);
     if (!file) throw new FileNotFoundError();
+    const model = FileService.getFileModel(baseFile);
+    const res = await model.findByIdAndUpdate(id, { $set: partialFile }, { runValidators: true }).exec();
 
-    const res = await fileModel.updateOne({ _id: file._id, ownerID: file.ownerID }, { $set: partialFile }, { runValidators: true }).exec();
-    // const res = await file.updateOne(partialFile, { runValidators: true }).exec();
-    return res.n === 1 && res.nModified === 1 && res.ok === 1;
+    return res.isModified();
   }
 
   /**
-   * Deletes a file from the DB.
+   * Delete a file from the DB.
    * @param id - the id of the file to be deleted.
    */
   static deleteById(id: string): Promise<IFile | null> {
     return fileModel.findByIdAndRemove({ _id: new ObjectID(id) }).exec();
   }
```

Comment:
What happens when it deletes a shortcut (what will it return)?

Reply:
done. 
made the func asynchronous and used the getFileModel method to get the correct model according to the file type.

Reply:
done